packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -152,13 +157,31 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
     void painter(Canvas canvas, Size size, double thumbOffset, double thumbExtent),
   ) {
     // Establish the minimum size possible.
-    double thumbExtent = math.min(viewport, minLength);
+    double thumbExtent = math.min(viewport, minOverscrollLength);
+
     if (before + inside + after > 0.0) {
+      // Thumb extent reflects fraction of content visible, if possible.
       final double fractionVisible = inside / (before + inside + after);
       thumbExtent = math.max(
         thumbExtent,
         viewport * fractionVisible - 2 * mainAxisMargin,
       );
+      // Thumb extent is no smaller than minLength if scrolling normally.
+      if (before != 0.0 && after != 0.0) {
+        thumbExtent = math.max(
+          minLength,
+          thumbExtent,
+        );
+      }
+      // Thumb extent is fraction of minLength if in overscroll. iOS behavior
+      // appears to be thumb reaching minimum size with ~20% of overscroll.
+      // Percentage of minLength maps from [0.8, 1.0] to [0.0, 1.0].
+      if (before == 0.0 || after == 0.0) {
+        thumbExtent = math.max(
+          thumbExtent,
+          minLength * (((inside / viewport) - 0.8) / 0.2),
```

Comment:
Doesn't seem to be necessarily the case. Try it with Settings -> General. We also shouldn't leak iOS behaviors into the widgets library.

Reply:
@willlarche: this was why I was asking you about the GoogleKit dogfoody app today. I can't find any examples of Google Material apps that doesn't use the native iOS scrollbar on iOS anymore. 

If Material itself never scrolls past the edges and we'll never use the non-iOS scrollbar even in Material when we do scroll past the edge, then maybe this is ok to be here.

Reply:
Checking on the iOS material catalog, it uses the standard iOS scrollbar too. Since only the iOS scrollbar scrolls past, I think it's ok here. Just make sure to document it.